bin/cheribsdtest/cheribsdtest_malloc.c

```diff
@@ -295,6 +295,29 @@ CHERIBSDTEST(malloc_revocation_ctl_suid_elfnote_enable_protctl_disable,
 	    "malloc_is_revoking_suid_elfnote_enable", true, &arg);
 }
 
+CHERIBSDTEST(malloc_early_constructor, "invoke malloc in an early constructor")
+{
+	int pid, res;
+
+	pid = fork();
+	CHERIBSDTEST_VERIFY(pid >= 0);
+	if (pid == 0) {
+		char progpath[] = "/usr/libexec/early_constructor";
```

Comment:
Why [] rather than *?

Reply:
Initially an arbitrary choice, but it turns you need to remove the const somewhere so I ended up leaving this. There's some argument these helpers should live in a a subdirectory of /usr/libexec and thus all the paths should be constructed with asprintf.